source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRHttpTarget.mm

```diff
@@ -0,0 +1,106 @@
+//
+//  ACRHttpTarget
+//  ACRHttpTarget.mm
+//
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import "ACRHttpTarget.h"
+#import "ACRContentHoldingUIView.h"
+#import "ACRIBaseInputHandler.h"
+#import "ACRViewController.h"
+
+@implementation ACRHttpTarget
+{
+    NSString *header;
+    NSString *body;
+    NSString *url;
+    NSString *method;
+    NSArray *_inputs;
+    __weak UIViewController *_vc;
+}
+
+- (instancetype)initWithHttpAction:(std::shared_ptr<AdaptiveCards::HttpAction> const&)httpAction
+                            inputs:(NSArray *)inputs
+                                vc:(UIViewController *)vc
+{
+    self = [super init];
+    if(self)
+    {
+        _vc = vc;
+        _inputs = inputs;
+        url = [NSString stringWithCString:httpAction->GetUrl().c_str() encoding:NSUTF8StringEncoding];
+        body = [NSString stringWithCString:httpAction->GetBody().c_str() encoding:NSUTF8StringEncoding];
+        method = [NSString stringWithCString:httpAction->GetMethod().c_str() encoding:NSUTF8StringEncoding];
+    }
+    return self;
+}
+
+- (NSString *)replaceMatchInString:(NSString *)string data:(NSDictionary *)data
+{
+    NSError *err = nil;
+    NSString *pattern = [[@"\\{\\{(" stringByAppendingString:[[data allKeys] componentsJoinedByString:@"|"]] stringByAppendingString:@")\\.Value\\}\\}"];
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:pattern options:0 error:&err];
+    __block NSMutableString *replacedString = [[NSMutableString alloc] init];
```

Comment:
>__block [](start = 4, length = 7)

This doesn't need to be a __block variable, since the reference isn't modified inside a block.